Mathlib/Data/ENat/Basic.lean

```diff
@@ -372,6 +372,12 @@ lemma add_right_injective_of_ne_top {n : ℕ∞} (hn : n ≠ ⊤) : Function.Inj
   simp_rw [add_comm n _]
   exact add_left_injective_of_ne_top hn
 
+lemma add_right_cancel_iff (a b c : ℕ∞) (netop : c ≠ ⊤) : a + c = b + c ↔ a = b :=
+  ⟨fun h ↦ ENat.add_left_injective_of_ne_top netop h, fun h ↦ by rw [h]⟩
+
+lemma add_left_cancel_iff (a b c : ℕ∞) (netop : c ≠ ⊤) : c + a = c + b ↔ a = b :=
+  ⟨fun h ↦ ENat.add_right_injective_of_ne_top netop h, fun h ↦ by rw [h]⟩
```

Comment:
```suggestion
lemma add_right_cancel_iff (a b : ℕ∞) {c : ℕ∞} (h_top : c ≠ ⊤) : a + c = b + c ↔ a = b :=
  ⟨fun h ↦ ENat.add_left_injective_of_ne_top netop h, fun h ↦ by rw [h]⟩

lemma add_left_cancel_iff (a b : ℕ∞) {c : ℕ∞} (h_top: c ≠ ⊤) : c + a = c + b ↔ a = b :=
  ⟨fun h ↦ ENat.add_right_injective_of_ne_top netop h, fun h ↦ by rw [h]⟩
```

Reply:
Actually, these are covered by `WithTop.add_right_inj` and `WithTop.add_left_inj`. Do you actually need these?